components/constellation/constellation.rs

```diff
@@ -1329,27 +1329,23 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
     fn handle_frame_size_msg(&mut self,
                              pipeline_id: PipelineId,
                              size: &TypedSize2D<f32, PagePx>) {
-        let msg = ConstellationControlMsg::Resize(pipeline_id, WindowSizeData {
-            visible_viewport: *size,
-            initial_viewport: *size * ScaleFactor::new(1.0),
-            device_pixel_ratio: self.window_size.device_pixel_ratio,
-        }, WindowSizeType::Initial);
-
-        // Store the new rect inside the pipeline
-        let result = {
-            // Find the pipeline that corresponds to this rectangle. It's possible that this
-            // pipeline may have already exited before we process this message, so just
-            // early exit if that occurs.
-            match self.pipelines.get_mut(&pipeline_id) {
-                Some(pipeline) => {
+        let result = match self.pipelines.get_mut(&pipeline_id) {
```

Comment:
The compiler complains about having mutably borrowed `self` outside the closure if I use this. Besides, I can't use map here because `result` can be `None` even if pipeline is resolved, in the case where sizes are equal.